flake.nix

```diff
@@ -0,0 +1,137 @@
+{
+  inputs = {
```

Comment:
So, Nix. Here are some thoughts; hopefully @robrix will chime in.

Nix certainly doesn’t seem to be going anywhere any time soon. If hosting our own Flake file makes it as easy as possible for Nix users to pull `fused-effects` into their project, I’m in favor of it. (If there is a good link showing the difference in workflow for a library with a flake.nix compared to one without one, I’d very much like to see it.) However, neither Rob or I use Nix, so our ability to maintain it is essentially zero. What I would need before accepting this would be:

1. one or more active Nix users to commit to maintaining it (also paging @jkachmar, muahaha); we’d be willing to hand out commit bits for that particular file;
2. CI integration via Actions (otherwise we have no way of knowing when it breaks)
3. a probationary period for a couple of GHC dot-releases: if Nix+Actions can keep up with .cabal in green-ness, then I’d like to keep them, but if they become one of those targets that’s perpetually red, then I’d rather excise it
4. ideally, some documentation on how to ensure version bumps and GHC upgrades go smoothly.